### PR TITLE
Command OpenContextMenu 

### PR DESCRIPTION
**Pull request by cinqmilleans**
**Resolved / Related Issues**
 * New command OpenContextMenu for open the context menu for the active element (item of layout, tab, ...).
 * Use the hotkey Ctrl+Enter.
 * Works for any interface element that has a ContextFlyout.
 * The homepage and sidebar elements don't have a ContextFlyout because the menu is generated by the RightTapped event. The action contains specific code for these 2 cases. To use this command on these elements, it is necessary to select them with the keyboard, and it is difficult to do. A pr to facilitate keyboard selection of homepage and sidebar elements is in preparation.
 * There is an incompatibility between hotkeys and flyouts. When a flyout is opened MainPage events are no longer fired. When you open the flyout with a hotkey, the keyreleased state is not modified when you release the key. MainPage contains a specific line of code to fix this. However, this problem may arise for other commands (OpenSettings, NewFolder, ...). A more permanent solution must be provided.

- [ ] Were these changes approved in an issue or discussion with the project maintainers? In order to prevent extra work, feature requests and changes to the codebase must be approved before the pull request will be reviewed. This prevents extra work for the contributors and maintainers.

**Validation**
How did you test these changes?
- [x] Did you build the app and test your changes?
- [x] Did you check for accessibility? You can use Accessibility Insights for this.
- [ ] Did you remove any strings from the en-us resource file?
   - [ ] Did you search the solution to see if the string is still being used? 
- [ ] Did you implement any design changes to an existing feature?
   - [ ] Was this change approved?
- [x] Are there any other steps that were used to validate these changes?
Test the hotkey Ctrl+Enter on item with contextmenu (selected by keyboard or mouse).

### CODE REVIEW

**Comment by yaira2** [comment]
I'm not convinced we need this, most keyboards have a menu key built for this purpose.

**Comment by cinqmilleans** [comment]
* This key is not present on my laptop, yet recent and professional. It is only available on my usb keyboard.
* This does not work in the sidebar and the homepage, because the menus are created dynamically by the RightTapped event, impossible to simulate on the keyboard (to my knowledge).
* The automatic placement by this extra key is strange, especially in the Detail view. The menu appears very to the right instead of being positioned at the beginning.
* The command will allow to have a personalized hotkey, including a single key. It is practical in case of handicap.

File: src/Files.App/Actions/Open/OpenContextMenuAction.cs [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+			var position = new Point(point.X + 16, point.Y + 16);
+
+			if (element.DataContext is INavigationControlItem)
+			{
+				var sidebar = element.FindAscendant<SidebarControl>();
+				if (sidebar is not null)
+				{
+					await sidebar.OpenContextMenuAsync(element, position);
+					return;
+				}
```
Comment (ferrariofilippo) [inline src/Files.App/Actions/Open/OpenContextMenuAction.cs:57]: You can remove the `return` here and also remove the brackets

File: src/Files.App/Actions/Open/OpenContextMenuAction.cs [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+			{
+				if (App.LastOpenedFlyout is CommandBarFlyout { IsOpen: true })
+					return false;
+
+				var element = GetFocusedElement();
+				if (element is null)
+					return false;
+
+				return element.ContextFlyout is FlyoutBase { IsOpen: false }
+					|| element.DataContext is INavigationControlItem or WidgetCardItem;
```
Comment (ferrariofilippo) [inline src/Files.App/Actions/Open/OpenContextMenuAction.cs:37]:
```suggestion
				return element.ContextFlyout is FlyoutBase { IsOpen: false } ||
					element.DataContext is INavigationControlItem or WidgetCardItem;
```

**Review by ferrariofilippo — COMMENTED** [review]
At the moment this shortcut is not working for tags widget and recent files. I think there's a conflict with another shortcut: pressing `Ctrl + Enter` will open items.

Also just got this exception when trying to open sidebar context menu (doesn't always happen)
[debug.log](https://github.com/files-community/Files/files/11071584/debug.log)

File: src/Files.App/Actions/Open/OpenContextMenuAction.cs (hunk 2 above)
Comment (cinqmilleans) [reply src/Files.App/Actions/Open/OpenContextMenuAction.cs:37]: I don't understand this change.

Comment (ferrariofilippo) [reply src/Files.App/Actions/Open/OpenContextMenuAction.cs:37]: In most of the codebase we align `&&`s and `||`s on the previous line

File: src/Files.App/Actions/Open/OpenContextMenuAction.cs (hunk 1 above)
Comment (cinqmilleans) [reply src/Files.App/Actions/Open/OpenContextMenuAction.cs]: It's done but I left the brackets, in accordance with the required style.

**Comment by cinqmilleans** [comment]
The fix for KeyReleased has been removed as it is no longer needed, thanks to a bug fix.

**Comment by cinqmilleans** [comment]
I fixed this command for RecentItems. There are still the tags to manage but it will be done in a day or two. Do not commit this pr.

**Comment by ferrariofilippo** [comment]
@cinqmilleans I was looking through the opened issues and I found this one #6085. You may consider implementing it

**Comment by yaira2** [comment]
@ferrariofilippo there are some pretty good alternatives in that issue so we're probably not going to implement that feature as originally requested.

File: src/Files.App/Actions/Open/OpenContextMenuAction.cs [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+using Windows.System;
+
+namespace Files.App.Actions
+{
+	internal class OpenContextMenuAction : IAction
+	{
+		public string Label { get; } = "OpenContextMenu".GetLocalizedResource();
+		public string Description => "TODO: Need to be described.";
+
+		public HotKey HotKey { get; } = new(VirtualKey.Enter, VirtualKeyModifiers.Control);
```
Comment (yaira2) [inline src/Files.App/Actions/Open/OpenContextMenuAction.cs:22]: Can you set the default to the `menu` key?

Comment (cinqmilleans) [reply src/Files.App/Actions/Open/OpenContextMenuAction.cs]: It's done.

**Comment by cinqmilleans** [comment]
The command now works in widgets. The hotkey is Menu+Enter.

File: src/Files.App/Actions/Open/OpenContextMenuAction.cs [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+using Windows.System;
+
+namespace Files.App.Actions
+{
+	internal class OpenContextMenuAction : IAction
+	{
+		public string Label { get; } = "OpenContextMenu".GetLocalizedResource();
+		public string Description => "TODO: Need to be described.";
+
+		public HotKey HotKey { get; } = new(VirtualKey.Enter, VirtualKeyModifiers.Menu);
```
Comment (yaira2) [inline src/Files.App/Actions/Open/OpenContextMenuAction.cs:23]: Can you remove the enter key?

Comment (yaira2) [reply src/Files.App/Actions/Open/OpenContextMenuAction.cs:23]: > Perhaps you mean the key that displays the menu

This is what I meant, and when we add support for remapping you'll be able to switch to Ctrl+Enter but the default should be the regular menu/application key.

Comment (cinqmilleans) [reply src/Files.App/Actions/Open/OpenContextMenuAction.cs:23]: This key is not present on all keyboards. Why not use the 2 default hotkeys ?

Comment (yaira2) [reply src/Files.App/Actions/Open/OpenContextMenuAction.cs:23]: Do you know of any other apps using Ctrl+Enter for this?

Comment (cinqmilleans) [reply src/Files.App/Actions/Open/OpenContextMenuAction.cs:23]: None. This is an opportunity to stand out.

Comment (yaira2) [reply src/Files.App/Actions/Open/OpenContextMenuAction.cs:23]: Can you add it to the list of shortcuts on the website?

Comment (cinqmilleans) [reply src/Files.App/Actions/Open/OpenContextMenuAction.cs:23]: I don't know enough node.js to do this quickly. Surely someone else can do this. Currently I am preparing a pr to modify hotkeys and store custom hotkeys in a litedatabase.

Comment (ferrariofilippo) [reply src/Files.App/Actions/Open/OpenContextMenuAction.cs:23]: Actually you have only to edit a markdown file (See https://github.com/files-community/Website/pull/413 for reference)

Comment (yaira2) [reply src/Files.App/Actions/Open/OpenContextMenuAction.cs:23]: > Currently I am preparing a pr to modify hotkeys and store custom hotkeys in a litedatabase.

Wouldn't json be better?

Comment (yaira2) [reply src/Files.App/Actions/Open/OpenContextMenuAction.cs:23]: This is out of scope for this pull request, the main thing is to update the file on the website documenation to include the shortcut.

Comment (cinqmilleans) [reply src/Files.App/Actions/Open/OpenContextMenuAction.cs]: I put the shortcuts Ctrl+Enter and the Application key.

**Comment by cinqmilleans** [comment]
Merge fixed.

**Pull request closed without merging**